tests/app/v2/notifications/test_post_notifications.py

```diff
@@ -362,6 +362,45 @@ def test_post_email_notification_returns_201(
     assert mock_deliver_email.called
 
 
+@pytest.mark.parametrize("reference", [None, "reference_from_client"])
+def test_post_email_notification_with_reply_to_returns_201(
+        client, sample_email_template_with_reply_to, mock_deliver_email, reference
+):
+    data = {
+        "email_address": sample_email_template_with_reply_to.service.users[0].email_address,
+        "template_id": sample_email_template_with_reply_to.id,
+        "personalisation": {"name": "Bob"},
+        "billing_code": "TESTCODE"
+    }
+    if reference:
+        data.update({"reference": reference})
```

Comment:
`data["reference"] = reference`

This isn't wrong, but there's no reason to use "update" for a single value.  Ditto for test_post_email_notification_returns_201.